src/main/scala/edu/ie3/osmogrid/lv/LvGraphBuilder.scala

```diff
@@ -0,0 +1,329 @@
+    *   for building connections
+    * @return
+    *   a Tuple of the distance and the point
+    */
+  private def getClosest(
+      wayNodeA: Node,
+      wayNodeB: Node,
+      buildingCenter: Coordinate,
+      minDistance: ComparableQuantity[Length]
+  ): Try[(ComparableQuantity[Length], Node)] = {
+    val coordinateA = buildCoordinate(wayNodeA.latitude, wayNodeA.longitude)
+    val coordinateB = buildCoordinate(wayNodeB.latitude, wayNodeB.longitude)
+    val orthogonalPt =
+      orthogonalProjection(coordinateA, coordinateB, buildingCenter)
+    // if orthogonal point is on the line and far enough apart from the line points
+    // take it since this is the closest we will get
+    if (
+      orthogonalPt.isBetween(
+        coordinateA,
+        coordinateB,
+        1e-3
+      ) && ((orthogonalPt haversineDistance coordinateA) isGreaterThan minDistance)
+      && ((orthogonalPt haversineDistance coordinateB) isGreaterThan minDistance)
+    ) {
+      val closestNode = Node(
+        id = UUID.randomUUID().getMostSignificantBits,
+        latitude = orthogonalPt.y,
+        longitude = orthogonalPt.x,
+        tags = Map.empty,
+        metaInformation = None
+      )
+      Success(buildingCenter.haversineDistance(orthogonalPt), closestNode)
+    }
+    // take the nearer point of the two line points
+    else {
+      val coordinateADistance = buildingCenter.haversineDistance(coordinateA)
+      val coordinateBDistance = buildingCenter.haversineDistance(coordinateB)
+      if (coordinateADistance.isLessThan(coordinateBDistance))
+        Success((coordinateADistance, wayNodeA))
+      else Success((coordinateBDistance, wayNodeB))
+    }
+  }
+
+  /** Updates the graph by adding the building graph connections and updating
+    * the edges of the surrounding nodes
+    *
+    * @param graph
+    *   the graph to update
+    * @param buildingGraphConnections
+    *   the building-graph connections to update with
+    * @return
+    *   the updated OsmGraph
+    */
+  private def updateGraphWithBuildingConnections(
+      graph: OsmGraph,
+      buildingGraphConnections: ParSeq[BuildingGraphConnection],
+      considerBuildingConnections: Boolean
+  ): OsmGraph = {
+    buildingGraphConnections.foreach(bgc => {
+      if (bgc.hasNewNode) {
+        graph.addVertex(bgc.graphConnectionNode)
+        graph.removeEdge(bgc.highwayNodeA, bgc.highwayNodeB)
+        graph.addWeightedEdge(bgc.highwayNodeA, bgc.graphConnectionNode)
+        graph.addWeightedEdge(bgc.graphConnectionNode, bgc.highwayNodeB)
+      }
+      if (considerBuildingConnections) {
+        val buildingNode = Node(
+          bgc.building.id,
+          bgc.center.y,
+          bgc.center.x,
+          Map.empty,
+          None
+        )
+        graph.addVertex(buildingNode)
+        graph.addWeightedEdge(bgc.graphConnectionNode, buildingNode)
+        bgc.copy(buildingNode = Some(buildingNode))
```

Comment:
Where is this copy used and how is it returned from the method?

Reply:
That is a very good catch 👍🏼  
Indeed that does nothing. We should update the building graph connections and return them, so that they can later be used for building the loads.